bin/automerge-prs

```diff
@@ -121,8 +121,8 @@ for json in "$tmp"/*.json; do
     *)
       printf '  \e[1;37m=>\e[0m \e[32mApprove and enable auto-merge\e[0m\n'
       if ((!DRY_RUN)); then
-        gh_pr review --approve "$number"
         gh_pr merge --auto "$number" --"$STRATEGY"
```

Comment:
If the Github setting of the repo doesn't allow auto merge, will it exit there and not run the approve part?

Reply:
This error (auto-merge not allowed in the repo) would make more sense to be shown to users as early as possible, no? Then we don't have to handle this case here

Reply:
> Then we don't have to handle [auto-merge not allowed in the repo] here

"auto-merge not allowed in the repo" is not what's being handled here. That was always an error before, and would remain an error now. Before, the approve would still happen (then error) whereas with this change, it would just error.

This change is addressing a race-condition that happens when our approve is the final requirement the PR is waiting for. When that's the case, and we approve first, the PR moves from the state where auto-merge can be enabled (because requirements have not been met yet) to the state where you can just merge it. GitHub does not allow you to enable auto-merge when the PR is in the latter state (called "clean status"). The only reason it works most of the time is because we do the two steps quickly enough that GitHub hasn't yet changed the state on its side. But when it does we hit the error.

Flipping the steps makes it so we enable auto-merge before we ourselves satisfy the PR requirements.